src/macros.cr

```diff
@@ -61,7 +61,13 @@
 # p.copy_with x: 3   # => #<Point(@x=3, @y=2)>
 # p                  # => #<Point(@x=0, @y=2)>
 # ```
-macro record(name, *properties)
+macro record(name, *properties, **kwargs)
+  {% raise "macro `record` does not accept named arguments
+    Did you mean:
+
+    record #{name}, #{(properties + kwargs.map{|name, value| "#{name} : #{value}"}).join(", ").id}
+    " unless kwargs.empty? %}
```

Comment:
Trailing condition here is pretty confusing.

```suggestion
  {% unless kwargs.empty? %}
    {% raise "macro `record` does not accept named arguments
      Did you mean:

      record #{name}, #{(properties + kwargs.map { |name, value| "#{name} : #{value}" }).join(", ").id}" %}
  {% end %}
```

Reply:
Heredoc reads even better IMO.
```suggestion
  {% raise <<-TXT unless kwargs.empty?
    macro `record` does not accept named arguments
    Did you mean:

    record #{name}, #{(properties + kwargs.map{|name, value| "#{name} : #{value}"}).join(", ").id}
    TXT
  %}
```

Reply:
Using heredocs triggers "cannot nest macro" error if the input is this:
```
record Cat: String
```

Might be related to the String type because that seeems to ignore the kwargs check and raises an undefined constant error.